learningresources/models.py

```diff
@@ -129,6 +130,29 @@ class LearningResource(BaseModel):
     xa_nr_attempts = models.IntegerField(default=0)
     xa_avg_grade = models.FloatField(default=0)
     xa_histogram_grade = models.FloatField(default=0)
+    url_name = models.TextField(null=True)
+
+    def get_preview_url(self):
+        """Create a preview URL"""
+        if self.url_name is not None:
+            url_format = ('{base_url}{org}/{course_number}'
+                          '/{run}/jump_to_id/{preview_id}')
+            return url_format.format(
+                base_url=LORE_PREVIEW_BASE_URL,
+                org=self.course.org,
+                course_number=self.course.course_number,
+                run=self.course.run,
+                preview_id=self.url_name,
+            )
+        else:
+            url_format = ('{base_url}{org}/{course_number}'
```

Comment:
I think we can leave this as is for now, but this first part `{org}/{course_number}/{run}/` breaks with SplitMongoModuleStore, and I don't think there is a way (without an API call), to find the right URL for a split course.  Do you happen to know @bdero if using the opaque key repo may help here to create a `CourseKey` from the course info?